### PR TITLE
Add substr() function

### PR DESCRIPTION
**Pull request by arodland**
Another aliasByNode-alike, but different.

### CODE REVIEW

File: expr.go [hunk 1]
```diff
@@ -2496,6 +2496,43 @@ func evalExpr(e *expr, from, until int32, values map[metricRequest][]*metricData
...
+		if err != nil {
+			return nil, err
+		}
+
+		startField, err := getIntNamedOrPosArgDefault(e, "start", 1, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		stopField, err := getIntNamedOrPosArgDefault(e, "stop", 2, 0)
```
Comment (nnuss) [inline expr.go]: Comment: by having >1 named argument this function will be subject to the same BUG mentioned in threshold().

File: expr.go [hunk 2]
```diff
@@ -2496,6 +2496,43 @@ func evalExpr(e *expr, from, until int32, values map[metricRequest][]*metricData
...
+		if err != nil {
+			return nil, err
+		}
+
+		fmt.Println(startField, stopField)
```
Comment (nnuss) [inline expr.go]: please remove this fmt.Println

File: expr.go [hunk 3]
```diff
@@ -2496,6 +2496,43 @@ func evalExpr(e *expr, from, until int32, values map[metricRequest][]*metricData
...
+
+		var results []*metricData
+
+		for _, a := range args {
+			metric := extractMetric(a.GetName())
+			nodes := strings.Split(metric, ".")
+			if startField != 0 {
```
Comment (nnuss) [inline expr.go]: startField and stopField should also check against len(nodes) to avoid "slice bounds out of range".

File: expr.go (hunk 2 above)
Comment (arodland) [inline expr.go]: eek, didn't mean to commit that obviously. :)

**Comment by nnuss** [comment]
Thanks @arodland, lgtm

**Pull request merged**